File: MenstrualApp.java
import java.util.Scanner;
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

public class MenstrualApp{

	public static void main(String... args){

		Scanner input = new Scanner(System.in);

		//System.out.print("Enter Age: ");
		//int age = input.nextInt();

		System.out.print("Enter Cycle Length(days): ");
		int cycleLength = input.nextInt();

		System.out.print("Enter Period Length(days): ");
		int periodLength  = input.nextInt();
		input.nextLine();

		System.out.print("Enter Last Period Start Date(yyyy-MM-dd): ");
		String lastPeriodInput = input.nextLine();
		System.out.println();

		LocalDate lastPeriod = lastPeriodStartDate(lastPeriodInput);

		LocalDate nextPeriodDate = nextPeriodStartDate(lastPeriod, cycleLength);

		LocalDate ovulationDate = ovulationDate(nextPeriodDate);

		System.out.println("======== USER'S REPORT ========");
		System.out.println("Next Period Date: " + nextPeriodDate);
		System.out.println("Ovulation Date: " + ovulationDate);
		System.out.println("Menstrual Phase: " + menstrualPhase(lastPeriod, periodLength));
		System.out.println("Fertile Window: " + fertileWindow(ovulationDate));
		System.out.println("Safe Periods: " + safePeriods(lastPeriod, nextPeriodDate, periodLength));


	}



	//Last Period Start Date
	public static LocalDate lastPeriodStartDate(String lastPeriodInput){

		DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
		LocalDate lastPeriodDate = LocalDate.parse(lastPeriodInput, formatter);

		return lastPeriodDate;

	}


	//Next Period Start Date
	public static LocalDate nextPeriodStartDate(LocalDate lastPeriod, int cycleLength){

		LocalDate nextPeriod = lastPeriod.plusDays(cycleLength);

		return nextPeriod;

	}


	//Ovulation Start Date
	public static LocalDate ovulationDate(LocalDate nextPeriod){

		LocalDate ovulationDate = nextPeriod.minusDays(14);

		return ovulationDate;

	}


	//Menstrual Phase
	public static String menstrualPhase(LocalDate lastPeriod, int periodLength){

		LocalDate endOfPeriod = lastPeriod.plusDays(periodLength - 1);

		String daysOfPeriod = lastPeriod + " to " + endOfPeriod;

		return daysOfPeriod;

	}


	//Fertile Window
	public static String fertileWindow(LocalDate ovulationDate){

		LocalDate startingDay = ovulationDate.minusDays(5);
		LocalDate endingDay = ovulationDate.plusDays(1);

		String fertilityWindow = startingDay + " to " + endingDay;

		return fertilityWindow;

	}


	//Safe Periods
	public static String safePeriods(LocalDate lastPeriod, LocalDate nextPeriod, int periodLength){

		LocalDate firstSafePeriodStartDay = lastPeriod;
		LocalDate firstSafePeriodEndDay = lastPeriod.plusDays(periodLength);

		LocalDate secondSafePeriodStartDay = nextPeriod.minusDays(8);
		LocalDate secondSafePeriodEndDay = nextPeriod;

		String safePeriods = firstSafePeriodStartDay + " to " + firstSafePeriodEndDay + " and " + secondSafePeriodStartDay + " to " + secondSafePeriodEndDay;

		return safePeriods;

	}







}

File: menstrual_app.py
from datetime import datetime, date, timedelta


# Last Period Start Date
def last_period_start_date(last_period_input):
	last_period_date = datetime.strptime(last_period_input, "%Y-%m-%d")
	return last_period_date


# Next Period Start Date
def next_period_start_date(last_period, cycle_length):
	next_period_date = last_period + timedelta(days=cycle_length)
	return next_period_date 


# Ovulation Date
def ovulation_date(next_period):
	ovulation_day = next_period - timedelta(days=14)
	return ovulation_day


# Menstrual Phase
def menstrual_phase(last_period, period_length):
	end_of_period = last_period + timedelta(days=period_length - 1)
	return f"{last_period.date()} to {end_of_period.date()}"


# Fertile Window
def fertile_window(ovulation):
	start_day = ovulation - timedelta(days=5)
	end_day = ovulation + timedelta(days=1)
	return f"{start_day.date()} to {end_day.date()}"


# Safe Periods
def safe_periods(last_period, next_period, period_length):
    first_safe_start = last_period
    first_safe_end = last_period + timedelta(days=period_length)

    second_safe_start = next_period - timedelta(days=8)
    second_safe_end = next_period

    return f"{first_safe_start.date()} to {first_safe_end.date()} and {second_safe_start.date()} to {second_safe_end.date()}"



cycle_length = int(input("Enter Cycle Length(days): "))
period_length = int(input("Enter Period Length(days): "))

last_period_input = input("Enter Last Period Start Date (yyyy-MM-dd): ")
print()

last_period = last_period_start_date(last_period_input)
next_period = next_period_start_date(last_period, cycle_length)
ovulation = ovulation_date(next_period)

print("======== USER'S REPORT ========")
print("Next Period Date:", next_period.date())
print("Ovulation Date:", ovulation.date())
print("Menstrual Phase:", menstrual_phase(last_period, period_length))
print("Fertile Window:", fertile_window(ovulation))
print("Safe Periods:", safe_periods(last_period, next_period, period_length))


